pkg/api/getambassador.io/v3alpha1/common.go

```diff
@@ -25,6 +25,42 @@ import (
 	"time"
 )
 
+// V2ExplicitTLS controls some vanity/stylistic elements when converting
+// from v3alpha1 to v2.  The values in an V2ExplicitTLS should not in any
+// way affect the runtime operation of Emissary; except that it may affect
+// internal names in the Envoy config, which may in turn affect stats
+// names.  But it should not affect any end-user observable behavior.
+type V2ExplicitTLS struct {
+	// TLS controls whether and how to represent the "tls" field when
+	// its value could be implied by the "service" field.
+	//
+	//  | Value        | Representation                        | Meaning of representation          |
+	//  |--------------+---------------------------------------+------------------------------------|
+	//  | ""           | omit the field                        | defer to service (no TLSContext)   |
+	//  | "null"       | store an explicit "null" in the field | defer to service (no TLSContext)   |
+	//  | "string"     | store an empty string in the field    | defer to service (no TLSContext)   |
```

Comment:
I thought `string` was a _non_-empty string. It'd be nice to maybe have this be `empty` or some such.

Reply:
A _non_-empty string means that there's a TLSContext, and the V2ExplicitTLS.TLS field is ignored.  This field is mostly "how to represent an 'empty' value" (except for `bool:true`, which is special).

Reply:
What about
```
	// TLS controls whether and how to represent the "tls" field when
	// its value could be implied by the "service" field.  In v2, there
	// were a lot of different ways to spell an "empty" value, and this
	// field specifies which way to spell it (if the value will indeed
	// be empty).
	//
	//  | Value        | Representation                        | Meaning of representation          |
	//  |--------------+---------------------------------------+------------------------------------|
	//  | ""           | omit the field                        | defer to service (no TLSContext)   |
	//  | "null"       | store an explicit "null" in the field | defer to service (no TLSContext)   |
	//  | "string"     | store an empty string in the field    | defer to service (no TLSContext)   |
	//  | "bool:false" | store a Boolean "false" in the field  | defer to service (no TLSContext)   |
	//  | "bool:true"  | store a Boolean "true" in the field   | originate TLS (no TLSContext)      |
```

Reply:
Yeah, that works.